### PR TITLE
[8.0] fixes for sql alchemy2.0

### PR DESCRIPTION
**Pull request by chaen**
closes https://github.com/DIRACGrid/DIRAC/issues/6737

BEGINRELEASENOTES

*DMS
CHANGE: adapt FTS3DB to sqlalchemy 2.0

*RMS
CHANGE: adapt ReqDB to sqlalchemy 2.0

ENDRELEASENOTES

### CODE REVIEW

**Review by chrisburr — DISMISSED** [review]

**Review by andresailer — CHANGES_REQUESTED** [review]
Also for 7.3 please!

**Comment by chrisburr** [comment]
Actually this should target v7r3

**Comment by chaen** [comment]
Grrrr I anyway have to make two PRs since `pyupgrade` makes conflict, so I will backport

**Comment by chrisburr** [comment]
Looks like there is something else:

> getRequest: unexpected exception : Strings are not accepted for attribute names in loader options; please use class-bound attributes directly

**Review by andresailer — DISMISSED** [review]
Documentation builds again :heavy_check_mark:

**Comment by chrisburr** [comment]
@chaen Do you understand why pylint keeps timing out?

**Comment by chaen** [comment]
I thought it was due to the problems github were having yesterday, but apparently not. I am investigating

File: pyproject.toml [hunk 1]
```diff
@@ -30,3 +27,37 @@ files = [
...
+
+[tool.pylint.main]
+# Files or directories to be skipped. They should be base names, not paths.
+ignore = [".svn", ".git", "integration_tests.py"]
+
+# List of module names for which member attributes should not be checked (useful
+# for modules/projects where namespaces are manipulated during runtime and thus
+# existing member attributes cannot be deduced by static analysis). It supports
+# qualified module names, as well as Unix pattern matching.
+ ignored-modules = ["MySQLdb", "numpy"]
```
Comment (chrisburr) [inline pyproject.toml]:
```suggestion
ignored-modules = ["MySQLdb", "numpy"]
```

Comment (chaen) [reply pyproject.toml]: thanks

File: src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py [hunk 2]
```diff
@@ -13,6 +13,8 @@
 from urllib.parse import unquote
 from functools import partial
 
+from typing import cast
```
Comment (fstagni) [inline src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py]: What's this for?

File: src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py [hunk 3]
```diff
@@ -333,8 +335,8 @@ def __pre_initialize(cls) -> list:
             cls.DEFAULT_LOCATION = cls._fullComponentName
 
         # SUPPORTED_METHODS should be a list type
-        if isinstance(cls.SUPPORTED_METHODS, str):
-            cls.SUPPORTED_METHODS = (cls.SUPPORTED_METHODS,)
+        if not isinstance(cls.SUPPORTED_METHODS, tuple):
```
Comment (fstagni) [inline src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py:336]: Correct the comment few lines above.

File: src/DIRAC/Core/Utilities/Pfn.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ def srm_pfnunparse(pfnDict):
             # host:port
             uri = "{}:{}".format(pfnDict["Host"], pfnDict["Port"])
         if pfnDict["WSUrl"]:
-            if "?" in pfnDict["WSUrl"] and "=" in pfnDict["WSUrl"]:
+            if "?" in pfnDict["WSUrl"] and "=" in pfnDict["WSUrl"]:  # pylint: disable = unsupported-membership-test
```
Comment (fstagni) [inline src/DIRAC/Core/Utilities/Pfn.py]:
```suggestion
            if "?" in pfnDict["WSUrl"] and "=" in pfnDict["WSUrl"]:  # pylint: disable=unsupported-membership-test
```

File: src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py (hunk 2 above)
Comment (chaen) [reply src/DIRAC/Core/Tornado/Server/private/BaseRequestHandler.py]: an attempt which failed. I'll remove

**Review by fstagni — APPROVED** [review]

**Pull request merged**